pkg/provider/loadbalancer/securitygroup/securitygroup.go

```diff
@@ -313,10 +313,11 @@ func (helper *RuleHelper) RemoveDestinationPrefixesFromRules(prefixes []string)
 	}
 
 	for _, rule := range helper.rules {
+		if rule.DestinationAddressPrefix != nil && index[*rule.DestinationAddressPrefix] {
+			rule.DestinationAddressPrefix = nil
+			continue
+		}
 		if rule.DestinationAddressPrefixes == nil {
```

Comment:
The original code does not work because `rule.DestinationAddressPrefixes != nil`? Is it an empty slice?

Reply:
yep